src/query/settings/src/settings_default.rs

```diff
@@ -290,6 +290,12 @@ impl DefaultSettings {
                     mode: SettingMode::Both,
                     range: Some(SettingRange::Numeric(0..=u64::MAX)),
                 }),
+                ("disable_merge_into_join_reorder", DefaultSettingValue {
```

Comment:
why not use `disable_join_reorder` directly?

Reply:
The `disable_merge_into_join_reorder` is used to disable the exchange between target and source, while `disable_join_reorder` is used to disable join reorder in source. @sundy-li

Reply:
Is it possible to add a  automatic check when to disable? Instead of adding a new setting by manual to disable it.